packages/components/CONTRIBUTING.md

```diff
@@ -4,12 +4,133 @@ Thank you for taking the time to contribute.
 
--   It has to be included in a `jsx` code block.
--   It has to work out-of-the-box. No additional code should be needed to have working the example.
--   It has to define a React component called `My<ComponentName>` which renders the example (i.e.: `MyButton`). Examples for the Higher Order Components should define a `MyComponent<ComponentName>` component (i.e.: `MyComponentWithNotices`).
+### Compatibility
+
+The `@wordpress/components` package includes several components which are relied upon by many developers across different projects. It is, therefore, very important to avoid introducing breaking changes as much as possible.
+
+In these situations, an alternative approach is to "soft-deprecate" the given legacy API. This is achieved by removing traces of the API from the docs, while the code keeps supporing it in the background (with or without a warning)
+
+### Components structure
+
+The contents of the `@wordpress/components` package can be divided into three groups: UI Primitives, UI Blocks, and Utilities.
+
+#### UI Primitives
+
+Lower-level components, which are usually composed together to build the second type of components: UI Blocks. UI Primitives usually don't contain translated text and aim at being highly configurable (at the expense of exposing more props and therefore more complexity).
+
+UI Primitives can be useful when in need of building an app with all new UI that looks and feels like WordPress but isn't powered by other WordPress technologies (like `@wordpress/i18n`).
+
+_[Note: we need to discuss how to properly classify/label these components. Should we introduce `primitives`, `blocks` and `utils` folders in the `src` folder? Should we include a standard way in each component's README to show which category it belongs to?]_
+
+#### UI Blocks
+
+Higher-level components, which usually offer fewer props compared to UI Primitives and contain translated text. These components use other WordPress technologies and are ready to be used in Gutenberg's UI.
+
+#### Utilities
+
+Components and other utilities function that don't necessarily render a piece of UI to screen, but are instead useful when implementing a given piece of functionality.
+
+### Components composition
+
+[To be expanded] E.g.:
+
+- Polymorphic Components
+- Using `children` vs custom render props vs arbitrary "data" props
+- Controlled and semi-controlled components
+- Composition patterns
+- Sub-components naming conventions
+- Components' layout responsibilities and boundaries (i.e., a component should only affect the layout of its children, not its own)
+- ...
+
+### APIs Consinstency
+
+[To be expanded] E.g.:
+
+- Boolean component props should be prefixed with `is*` (e.g. `isChecked`) or `has*` (`hasValue`).
+- Event callback props should be prefixed with `on*` (e.g. `onChanged`)
+- `styled` components should be suffixed with `*View` (e.g. `BoxView`)
```

Comment:
This seems like a bad guideline to me now. `const RangeControlView = styled( RangeControl )` doesn't make any sense to me. In what sense is it a "View"?

Reply:
I was under the impression that this pattern had been agreed in previous PRs as a way to recognise `styled` components — here's an [example](https://github.com/WordPress/gutenberg/pull/33054/files#diff-3ba4eeae54a94e8a8f295d9d9052048a33ef16f5f69880ac69dc4dd6138dd752R28).

Anyway, I removed this line in [`aea7c60` (#33960)](https://github.com/WordPress/gutenberg/pull/33960/commits/aea7c603b2ad7f1cca740d667cb1dcab273a84cc)

Reply:
Right, I think we'd discussed it, I'm just rechallenging it after more experience with it before it becomes codified 🙂